### PR TITLE
Fix - SVM account_loader tests

### PR DESCRIPTION
**Pull request by Lichtso**
#### Problem
Some tests in svm/src/account_loader.rs currently do test situations which are impossible to construct in production and do miss cases which are. The issue is a slightly incorrect setup of the program accounts involved. The tests assume that the program cache of the program cache is protocol relevant, which it is not, instead the SVM program loader is.

Otherwise these tests (introduced in #3045) are pretty good, so shout-out to Hana for writing them!

#### Summary of Changes
- Fixes `test_load_transaction_accounts_program_account_executable_bypass`:
Now constructs a loader-v2 tombstone, instead of an impossible program cache entry (a closed built-in) which would not be produced by the SVM program loader. This also changes the expected owner of the loaded program account from the `native_loader` to `bpf_loader` and thus inserts `bpf_loader` in the list of loaded accounts.

- Fixes `test_load_transaction_accounts_data_sizes`:
Now initializes the loader-v3 program with a valid `UpgradeableLoaderState::Program` pointing to a programdata account containing `UpgradeableLoaderState::ProgramData` and an ELF. Before the test simply forced in an impossible program cache entry (a closed built-in) which would not be produced by the SVM program loader. Also, uses the correctly initialized program cache for all tests now. Before it had some with an empty cache, which again can't happen in production because the cache is replenished before the account loader runs. Furthermore, adds two test cases for write lock demotion of loader-v3.

- Removes `test_load_transaction_accounts_program_account_not_found_but_loaded`:
The test constructs a scenario in which the program cache contains an entry even tough the accounts database has no account at the given key. This is not possible because the program cache only has entries for accounts owned by the loaders. A non existent account can not be owned by a loader.

Fixes #

### CODE REVIEW

**Review by 2501babe — DISMISSED** [review]

**Comment by Lichtso** [comment]
I think we are still missing coverage for the distinction in the size of valid program entries and tombstones in `test_load_transaction_accounts_data_sizes`. I will add that as well.

**Comment by Lichtso** [comment]
@2501babe can you take a look again? I changed the loader-v2 program to be closed so it counts as 0 and the loader-v3 program to be loaded correctly. Furthermore, I added test coverage for when loader-v3 experiences write lock demotion which affects if its programdata is counted.

File: svm/src/account_loader.rs [hunk 1]
```diff
@@ -2648,22 +2618,28 @@ mod tests {
...
             )];
-            test_data_size_with_cache(
+            test_data_size(
                 ixns,
-                program_cache.clone(),
                 program2_size + upgradeable_loader_size + fee_payer_size,
             );
 
-            // programdata as instruction account double-counts it
+            // programdata as readonly instruction account double-counts it
```
Comment (2501babe) [inline svm/src/account_loader.rs:2626]:
```suggestion
            // programdata as instruction account double-counts it
```

these tests are in a loop that use both read-only and writable account metas

**Review by 2501babe — APPROVED** [review]

File: svm/src/account_loader.rs (hunk 1 above)
Comment (Lichtso) [reply svm/src/account_loader.rs:2626]: yep, but only one case is double counted. See `factor` below

**Pull request merged**